python/test/unit/cuda/test_experimental_tma.py

```diff
@@ -227,3 +227,45 @@ def test_device_tensormap1d(dtype_str):
     # Check results are correct
     torch.testing.assert_close(unwrap_tensor(inp), unwrap_tensor(out))
     torch.testing.assert_close(unwrap_tensor(inp), unwrap_tensor(inp_copy))
+
+
+"""
+Regression test: descriptor store should implicitly downcast values to match
+descriptor element type. For float16/bfloat16 descriptors, storing a float32
+block should succeed and round-trip downcast.
+"""
+
+
+@requires_tma
+@pytest.mark.interpreter
+@pytest.mark.parametrize("dtype_str", ["float16", "bfloat16"])
+def test_tensor_descriptor_store_downcast(dtype_str):
+
+    @triton.jit
+    def kernel(out_ptr, M, N, M_BLOCK: tl.constexpr, N_BLOCK: tl.constexpr):
+        moffset = tl.program_id(axis=0) * M_BLOCK
+        noffset = tl.program_id(axis=1) * N_BLOCK
+        midx = moffset + tl.arange(0, M_BLOCK)[:, None]
+        nidx = noffset + tl.arange(0, N_BLOCK)[None, :]
+        # Generate a float32 block based on linear indices.
+        val_f32 = (midx * N + nidx).to(tl.float32)
+        desc = tl.make_tensor_descriptor(
+            out_ptr,
+            shape=[M, N],
+            strides=[N, 1],
+            block_shape=[M_BLOCK, N_BLOCK],
+        )
+        desc.store([moffset, noffset], val_f32)
```

Comment:
you appear to have forgotten to add a runtime memory allocator using `triton.set_allocator`. Please see other examples in the unit tests for how to add one